Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -487,4 +560,54 @@ namespace AZ::Render
             auxGeom->DrawLines(lineArgs);
         }
     }
+
+    void AtomActorDebugDraw::RenderJointNames(EMotionFX::ActorInstance* actorInstance,
+        RPI::ViewportContextPtr viewportContext, const AZ::Color& jointNameColor)
+    {
+        if (!m_fontDrawInterface)
+        {
+            auto fontQueryInterface = AZ::Interface<AzFramework::FontQueryInterface>::Get();
+            if (!fontQueryInterface)
+            {
+                return;
+            }
+            m_fontDrawInterface = fontQueryInterface->GetDefaultFontDrawInterface();
+        }
+
+        if (!m_fontDrawInterface || !viewportContext || !viewportContext->GetRenderScene() ||
+            !AZ::Interface<AzFramework::FontQueryInterface>::Get())
+        {
+            return;
+        }
+
+        const EMotionFX::Actor* actor = actorInstance->GetActor();
+        const EMotionFX::Skeleton* skeleton = actor->GetSkeleton();
+        const EMotionFX::TransformData* transformData = actorInstance->GetTransformData();
+        const EMotionFX::Pose* pose = transformData->GetCurrentPose();
+        const size_t numEnabledNodes = actorInstance->GetNumEnabledNodes();
+
+        m_drawParams.m_drawViewportId = viewportContext->GetId();
+        auto viewportSize = viewportContext->GetViewportSize();
+        m_drawParams.m_position = AZ::Vector3(static_cast<float>(viewportSize.m_width), 0.0f, 1.0f) +
+            TopRightBorderPadding * viewportContext->GetDpiScalingFactor();
+        m_drawParams.m_color = jointNameColor;
+        m_drawParams.m_scale = AZ::Vector2(BaseFontSize);
+        m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Right;
+        m_drawParams.m_monospace = false;
+        m_drawParams.m_depthTest = false;
+        m_drawParams.m_virtual800x600ScreenSize = false;
+        m_drawParams.m_scaleWithWindow = false;
```

Comment:
Do we need to set all of these? Some look like default values?

Reply:
Not really but we also don't want to depend on the default value